android/app/src/main/java/com/mattermost/helpers/CustomPushNotificationHelper.java

```diff
@@ -227,6 +238,153 @@ public static void createNotificationChannels(Context context) {
         }
     }
 
+    public static boolean verifySignature(final Context context, String signature, String serverUrl, String ackId) {
+        if (signature == null) {
+            Log.i("Mattermost Notifications Signature verification", "No signature in the notification");
+            return true;
+        }
+
+        if (serverUrl == null) {
+            Log.i("Mattermost Notifications Signature verification", "No server_url for server_id");
+            return false;
+        }
+
+        DatabaseHelper dbHelper = DatabaseHelper.Companion.getInstance();
+        if (dbHelper == null) {
+            Log.i("Mattermost Notifications Signature verification", "Cannot access the database");
+            return false;
+        }
+
+        WMDatabase db = getDatabaseForServer(dbHelper, context, serverUrl);
+        if (db == null) {
+            Log.i("Mattermost Notifications Signature verification", "Cannot access the server database");
+            return false;
+        }
+
+        if (signature.equals("NO_SIGNATURE")) {
+            String version = queryConfigServerVersion(db);
+            if (version == null) {
+                Log.i("Mattermost Notifications Signature verification", "No server version");
+                return false;
+            }
+
+            if (!version.matches("[0-9]+(\\.[0-9]+)*")) {
+                Log.i("Mattermost Notifications Signature verification", "Invalid server version");
+                return false;
+            }
+
+            String[] parts = version.split("\\.");
+            int major = parts.length > 0 ? Integer.parseInt(parts[0]) : 0;
+            int minor = parts.length > 1 ? Integer.parseInt(parts[1]) : 0;
+            int patch = parts.length > 2 ? Integer.parseInt(parts[2]) : 0;
+
+            int[][] targets = {{9,8,0},{9,7,2},{9,6,2},{9,5,4},{8,1,13}};
+            boolean rejected = false;
+            for (int i = 0; i < targets.length; i++) {
+                boolean first = i == 0;
+                int[] targetVersion = targets[i];
+                int majorTarget = targetVersion[0];
+                int minorTarget = targetVersion[1];
+                int patchTarget = targetVersion[2];
+
+                if (major > majorTarget) {
+                    // Only reject if we are considering the first (highest) version.
+                    // Any version in between should be acceptable.
+                    rejected = first;
+                    break;
+                }
+
+                if (major < majorTarget) {
```

Comment:
Should equal value case also be included here?

Reply:
We only check the minor version if the version is exactly the same. So... imagine we have the targets `9.8.0` and `7.5.0`.

If we are on version `8.0.0`, when comparing with `9.8.0`, since the major version is lower, we can continue to the next target. But if we are on version `9.9.0`, even if the major version is the same, we want to check the minor version.

I hope it makes sense